spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Optimizer.scala

```diff
@@ -314,7 +330,17 @@ abstract class Optimizer[T: ClassTag, D](
    * @param method optimization method
    */
   def setOptimMethod(method : OptimMethod[T]): this.type = {
-    this.optimMethod = method
+    this.optimMethods = Map(model.getName -> method)
+    this
+  }
+
+  /**
+   * Set optimization methods for each submodule.
+   *
+   * @param method A mapping of submodule -> OptimMethod
+   */
+  def setOptimMethods(method: Map[String, OptimMethod[T]]): this.type = {
```

Comment:
You should override this in the LocalOptimizer to throw exception

Reply:
I have implement multi optimMethods for localOptimizer